met_connectors/src/frost/series.rs

```diff
@@ -1,24 +1,12 @@
-use crate::frost::{duration, Error, FrostObs};
+use crate::frost::{duration, Error, FrostLatLonElev, FrostObs};
 use chrono::{prelude::*, Duration};
 use chronoutil::RelativeDuration;
-use rove::data_switch::{self, SeriesCache, Timerange, Timestamp};
+use rove::data_switch::{self, DataCache, Timerange, Timestamp};
 
-fn extract_duration(mut metadata_resp: serde_json::Value) -> Result<RelativeDuration, Error> {
-    let time_resolution = metadata_resp
-        .get_mut("data")
-        .ok_or(Error::FindMetadata(
-            "couldn't find data field on root".to_string(),
-        ))?
-        .get_mut("tseries")
-        .ok_or(Error::FindMetadata(
-            "couldn't find field tseries on data".to_string(),
-        ))?
-        .get_mut(0)
-        .ok_or(Error::FindMetadata("tseries array is empty".to_string()))?
-        .get_mut("header")
-        .ok_or(Error::FindMetadata(
-            "couldn't find field header on 1st tseries".to_string(),
-        ))?
+use super::spatial::extract_location;
+
+pub(crate) fn extract_duration(header: &mut serde_json::Value) -> Result<RelativeDuration, Error> {
```

Comment:
Since this is now used in the spatial module too, rather than it remaining here and adding `pub(crate)`, I think we should put it in a separate module (mod util, perhaps?) that both import. The same applies to `extract_location`.